### ISSUE TITLE
502 error when login behind reverse proxy

### ISSUE THREAD

**Issue by FloLaco** [keycloak/keycloak#13939]
### Describe the bug

Hi,

I've deployed keycloak on a Kube cluster with nginx ingress with proxy edge mode specified.
Between ingress proxy and KC, it's plain HTTP and I'm sending traffic to port 8080.
I'm accessing to admin console as expected.
From my application, when I click to "Login", I'm well redirected to keycloak login page. I'm typing a login/pwd and then, when I click on "Sign in", I'm getting the 502 error 

![image](https://user-images.githubusercontent.com/1486562/186208683-09e27c28-50db-4769-9978-26bdba4779c2.png)

(on chrome console : POST https://XXX.com/realms/myrealm/login-actions/authenticate?session_code=lOYKhNaPJCwg1f7FkWPmMbpEGrCBTPwn623HsudFzcI&execution=c9228b31-41e9-4bed-a544-bc9a4bd0d30c&client_id=myclient&tab_id=zH78yZd5irs )

### Version

19.0.1

**Comment by pedroigor** [comment]
@FloLaco Did you check the server logs for any error and the status code returned by the server to your proxy?

**Comment by FloLaco** [comment]
@pedroigor I did not see any log on keycloak pod, even in debug verbose log.
Hiw can I check the status code returned to proxy without accessing to proxy log (because it's a blackbox, cannot access to ingress logs)

**Comment by pedroigor** [comment]
If you already tried setting the `debug` level, I don't have any other sugestion.

Perhaps you can look at the access logs by enabling this Quarkus property: https://quarkus.io/guides/http-reference#configuring-http-access-logs. See [Using raw Quarkus Properties](https://www.keycloak.org/server/configuration).

**Comment by FloLaco** [comment]
@pedroigor Thanks for your answer.
You help me to fix the issue. 

After activating quarkus access log, I saw that KC replied a 302 but I saw a 502. So KC was not the issue, but the Kubernetes Ingress.
`2022-08-24 08:02:30,746 INFO  [io.quarkus.http.access-log] (executor-thread-1) 10.54.74.54 - - 24/Aug/2022:08:02:30 +0000 "POST /realms/myrealm/login-actions/authenticate?session_code=KMcMj5dz5EBYKk2_Oh5zQhsfQFaQeuqbrsGdfNJZDps&execution=c9228b31-41e9-4bed-a544-bc9a4bd0d30c&client_id=myclient&tab_id=y9G8ZPM-MyY HTTP/1.1" 302 -`

With big luck, I found the issue by type 'ingress nginx security 502 bad gateway redirect' on Google : https://stackoverflow.com/questions/58943111/nginx-ingress-returns-502-after-post-with-redirect
I just had to add this annotations to my ingress deployment : 
```
    nginx.ingress.kubernetes.io/proxy-buffer-size: "64k"
    nginx.ingress.kubernetes.io/proxy-buffers-number: "8"
```
